lua/lvim/core/autocmds.lua

```diff
@@ -34,6 +34,59 @@ function M.load_defaults()
         end,
       },
     },
+    {
+      "BufReadPre",
+      {
+        group = "_general_settings",
+        pattern = "*",
+        desc = "Performance improvements for big files",
+        callback = function(args)
+          if vim.b.big_file then
+            return
+          end
+          local MB = 1024 * 1024
+          local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(args.buf))
+          if not (ok and stats and stats.size > MB) then
+            return
+          end
+          vim.b.big_file = true
+          vim.g.big_files = vim.g.big_files and vim.g.big_files + 1 or 1
```

Comment:
generic name as well, also, is this really needed?

Reply:
it's needed for running 
```
vim.cmd "DoMatchParen"
vim.cmd "syntax on"
```
on BufDelete so it doesn't run in case if more than one big file is open